typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,592 @@
+                if (error === null) {
+                    resolve(okAsync(response));
+                } else {
+                    resolve(errAsync(error));
+                }
+            })
+        });
+        const resultRegisterStaticFiles: Result<RegisterStaticFilesResponse, Error> = await promiseRegisterStaticFiles;
+        if (!resultRegisterStaticFiles.isOk()) {
+            return err(resultRegisterStaticFiles.error);
+        }
+        const resp: RegisterStaticFilesResponse = resultRegisterStaticFiles.value;
+
+        const staticFileDestRelativeFilepathsMap: Map<string, string> = resp.getStaticFileDestRelativeFilepathsMap();
+        for (let [staticFileIdStr, destFilepathRelativeToEnclaveVolRoot] of staticFileDestRelativeFilepathsMap.entries()) {
+
+            const staticFileId: StaticFileID = <StaticFileID>(staticFileIdStr);
+
+            if (!staticFileFilepaths.has(staticFileId)) {
+                return err(new Error("No source filepath found for static file " + staticFileId + "; this is a bug in Kurtosis"));
+            }
+            const srcAbsFilepath: string = staticFileFilepaths[staticFileId];
+
+            const destAbsFilepath: string = path.join(this.enclaveDataVolMountpoint, destFilepathRelativeToEnclaveVolRoot);
+            
+            const promiseStatDestAbsFilepath: Promise<ResultAsync<fs.Stats, Error>> = new Promise((resolve, _unusedReject) => {
+                fs.stat(destAbsFilepath, (error: Error, response: fs.Stats) => {
+                    if (error === null) {
+                        resolve(okAsync(response));
+                    } else {
+                        resolve(errAsync(error));
+                    }
+                })
+            });
+            const resultStatDestAbsFilepath: Result<fs.Stats, Error> = await promiseStatDestAbsFilepath;
+            if (!resultStatDestAbsFilepath.isOk()) {
+                return err(new Error("The Kurtosis API asked us to copy static file " + staticFileId + " to path " + destFilepathRelativeToEnclaveVolRoot + 
+                " in the enclave volume which means that an empty file should exist there, " + "but no file exists at that path - this is a bug in Kurtosis!"));
+            }
+    
+            const promiseOpenSrcFp: Promise<ResultAsync<number, Error>> = new Promise((resolve, _unusedReject) => {
+                fs.open(srcAbsFilepath, 'r', (error: Error, response: number) => {
+                    if (error === null) {
+                        resolve(okAsync(response));
+                    } else {
+                        resolve(errAsync(error));
+                    }
+                })
+            });
+            const resultOpenSrcFp: Result<number, Error> = await promiseOpenSrcFp;
+            if (!resultOpenSrcFp.isOk()) {
+                return err(resultOpenSrcFp.error);
+            }
+            let srcFp: number;
```

Comment:
this should be `const srcFp: number = resultOpenSrcFp.value` - there's no need to make it mutable with `let`, or to do the assignment in the line below inside the `try` statement. By splitting it up, it opens the door for someone to add extra logic on line 175, which - if it errors - will yet again cause an NPE when the `finally` tries to close a null `srcFp`. 

Gotta think really carefully about these things, making sure that there's no way to silently NPE!!